header-rewrite-filter/header_processor_test.cc

```diff
@@ -127,33 +127,41 @@ TEST_F(ProcessorTest, SetPathProcessorTest) {
 
 TEST_F(ProcessorTest, SetBoolProcessorTest) {
     std::vector<absl::string_view> true_match_test_cases = {
-        "http set-bool mock_bool matches -m str matches", // exact
-        "http set-bool mock_bool matches -m beg ma", // prefix
-        "http set-bool mock_bool matches -m sub tch", // substring
+        "http set-bool mock_bool hdr(mock_header1) -m str mock_value3", // exact, hdr 1 arg
+        "http set-bool mock_bool hdr(mock_header1,-1) -m str mock_value3", // exact, hdr 2 arg
+        "http set-bool mock_bool hdr(mock_header1,0) -m str mock_value1", // exact, hdr 2 arg
+        "http set-bool mock_bool hdr(mock_header1,-3) -m str mock_value1", // exact, hdr 2 arg
+        "http set-bool mock_bool hdr(mock_header2) -m beg mo", // prefix
+        "http set-bool mock_bool urlp(param1) -m beg so", // prefix, urlp
+        "http set-bool mock_bool hdr(mock_header2) -m sub lue", // substring
+        "http set-bool mock_bool hdr(mock_header2) -m found", // found
+        "http set-bool mock_bool urlp(param2) -m found", // found, urlp
```

Comment:
Can you add a test e.g.

```
http set-bool mock_bool urlp(param2 -m found)
```
? https://github.com/DataDog/envoy-filter-example/pull/26/files#diff-4b242b3f6b47b7508727b3a8a9bf70424b85329968fb32a957ba9a2cb575cdc7R62 might be problematic (last index)